src/int/mul.rs

```diff
@@ -31,13 +31,30 @@ impl<const LIMBS: usize> Int<LIMBS> {
 
         (lo, hi, negate)
     }
+
+    /// Multiply `self` by `rhs`, returning a concatenated "wide" result.
+    pub const fn widening_mul<const RHS_LIMBS: usize, const WIDE_LIMBS: usize>(
+        &self,
+        rhs: &Int<RHS_LIMBS>,
+    ) -> Int<WIDE_LIMBS>
+    where
+        Uint<LIMBS>: ConcatMixed<Uint<RHS_LIMBS>, MixedOutput = Uint<WIDE_LIMBS>>,
+    {
+        let (lhs_abs, lhs_sign) = self.abs_sign();
+        let (rhs_abs, rhs_sign) = rhs.abs_sign();
+        let product_abs = lhs_abs.widening_mul(&rhs_abs);
+        let product_sign = lhs_sign.xor(rhs_sign);
+
+        // always fits
+        Int::new_from_uint(product_abs.wrapping_neg_if(product_sign))
```

Comment:
I agree that it should always fit. However, why not use `Int::from_abs_sgn(product_abs, product_sign).unwrap()`?

Reply:
unwrap is not `const fn`

Reply:
Ah, good point!